src/util/helpers.tsx

```diff
@@ -262,6 +261,37 @@ export const getAbsoluteHeightBelow = (belowId: string): number => {
   return element.offsetHeight + margin + padding + 1;
 };
 
+export const getAbsoluteHeightBelow = (belowId: string): number => {
+  const element = belowId ? document.getElementById(belowId) : undefined;
+  if (!element) {
+    return 0;
+  }
+  return getElementAbsoluteHeight(element);
+};
+
 export const generateUUID = (): string => {
   return crypto.randomBytes(16).toString("hex");
 };
+
+export const getClientOS = (userAgent: string) => {
+  if (userAgent.includes("Windows")) {
+    return "windows";
+  } else if (userAgent.includes("Mac OS")) {
+    return "macos";
+  } else if (userAgent.includes("Linux")) {
+    return "linux";
+  }
+
+  return null;
+};
+
+export const generateUniqueArray = <T,>(array: T[]) => {
+  return Array.from(new Set(array));
+};
+
+export const isMobileBrowser = () => {
+  const userAgent = navigator.userAgent;
+  return /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(
+    userAgent,
+  );
+};
```

Comment:
This is probably not the best way to do it, better use `@media screen and (pointer: fine)`? Because the browser doesn't tell you if the user is on a touch screen or has connected a pointing device.